src/app/chart/chart.component.ts

```diff
@@ -1,20 +1,36 @@
-import { Component, OnInit } from '@angular/core';
+import {
+  Component,
+  Input,
+  OnChanges,
+  SimpleChanges,
+  OnInit,
+} from '@angular/core';
 import { MockDataService } from '../mockData/Table data';
 
 @Component({
   selector: 'app-chart',
   templateUrl: './chart.component.html',
-  styleUrl: './chart.component.scss',
+  styleUrls: ['./chart.component.scss'],
 })
-export class ChartComponent implements OnInit {
+export class ChartComponent implements OnInit, OnChanges {
+  @Input() data: any[] = [];
   ordersData: any;
   chartData: any;
 
   constructor(public mockDataService: MockDataService) {}
 
   ngOnInit(): void {
-    this.ordersData = this.mockDataService
-      .getData()
+    this.updateChartData();
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes['data'] && !changes['data'].firstChange) {
+      this.updateChartData();
+    }
+  }
+
```

Comment:
Refactor **ChartComponent** to remove **OnInit** and **OnChanges** lifecycle hooks. Instead use a setter on the @Input() data property to directly trigger updateChartData method whenever 'data' changes. 
This approach simplifies the component and reduces lifecycle complexity.